pkg/disk/partition.go

```diff
@@ -18,6 +19,44 @@ type Partition struct {
 	Payload Entity
 }
 
+// XXX: this needs refactoring and tests, it's kinda complicated
```

Comment:
I will work on this in a followup/parallel PR, this is tested indirectly via the otk-make-* tests but of course a direct test is much better and will get added. Note that this is only used by the otk-make-* tooling right now so the risk to merge in the current state is low.

Reply:
I started this in https://github.com/osbuild/images/pull/737